charts/catalog/templates/apiserver-deployment.yaml

```diff
@@ -0,0 +1,139 @@
+kind: Deployment
+apiVersion: extensions/v1beta1
+metadata:
+  name: {{ template "fullname" . }}-apiserver
+  labels:
+    app: {{ template "fullname" . }}-apiserver
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ template "fullname" . }}-apiserver
+  template:
+    metadata:
+      labels:
+        app: {{ template "fullname" . }}-apiserver
+        chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+        release: "{{ .Release.Name }}"
+        heritage: "{{ .Release.Service }}"
+    spec:
+      containers:
+      - name: apiserver
+        image: {{ .Values.apiserver.image }}
+        imagePullPolicy: {{ .Values.apiserver.imagePullPolicy }}
+        resources:
+          requests:
+            cpu: 100m
+            memory: 20Mi
+          limits:
+            cpu: 100m
+            memory: 30Mi
+        args:
+        - --secure-port
+        - "8443"
+        {{- if .Values.apiserver.insecure }}
+        - --insecure-bind-address
+        - 0.0.0.0
+        - --insecure-port
+        - "8080"
+        {{- end }}
+        - --storage-type
+        - {{ .Values.apiserver.storage.type }}
+        {{- if eq .Values.apiserver.storage.type "etcd" }}
+        - --etcd-servers
+        - {{ .Values.apiserver.storage.etcd.servers }}
+        {{- else if eq .Values.apiserver.storage.type "tpr" }}
+        - --global-namespace
+        - {{ .Values.apiserver.storage.tpr.globalNamespace }}
+        {{- end }}
+        - -v
+        - "{{ .Values.apiserver.verbosity }}"
+        ports:
+        {{- if .Values.apiserver.insecure }}
+        - containerPort: 8080
+        {{- end }}
+        - containerPort: 8443
+        volumeMounts:
+        - name: apiserver-cert
+          mountPath: /var/run/kubernetes-service-catalog
+          readOnly: true
+        readinessProbe:
+          httpGet:
+            port: 8443
+            path: /healthz
+            scheme: HTTPS
```

Comment:
Running insecure is additive. It doesn't eliminate the secure endpoint. And the health checks don't care about cert validity... so on account of all that, this is ok... and I have manually verified.